AsyncDisplayKit/ASNavigationController.m

```diff
@@ -0,0 +1,91 @@
+//
+//  ASNavigationController.m
+//  Pods
+//
+//  Created by Garrett Moon on 4/27/16.
+//
+//
+
+#import "ASNavigationController.h"
+
+@implementation ASNavigationController
+{
+  BOOL _parentManagesVisibilityDepth;
+  NSInteger _visibilityDepth;
+}
+
+ASVisibilityDidMoveToParentViewController;
+
+ASVisibilityViewWillAppear;
+
+ASVisibilityViewDidDisappearImplementation;
+
+ASVisibilityDepthImplementation;
+
+- (void)visibilityDepthDidChange
+{
+  for (UIViewController *viewController in self.viewControllers) {
+    if ([viewController conformsToProtocol:@protocol(ASVisibilityDepth)]) {
+      [(id <ASVisibilityDepth>)viewController visibilityDepthDidChange];
+    }
+  }
+}
+
+- (NSInteger)visibilityDepthOfChildViewController:(UIViewController *)childViewController
+{
+  NSUInteger viewControllerIndex = [self.viewControllers indexOfObject:childViewController];
+  NSAssert(viewControllerIndex != NSNotFound, @"childViewController is not in the navigation stack.");
+  
+  if (viewControllerIndex == self.viewControllers.count - 1) {
+    //view controller is at the top
+    return [self visibilityDepth] + 0;
```

Comment:
I thought it a bit more readable, but maybe it's more confusing than clarifying?